torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -318,55 +291,151 @@ class AttributePropagator {
     }
   }
 
-  // cleanupFrozenModule function cleans up the Frozen module. it performs the
+  bool moduleEscapes(Module& subModule, std::shared_ptr<Graph>& graph) {
```

Comment:
This doesn't check nested types. I think you want to recurse through the contained types of the graph outputs. Also I believe it is an invariant that jit graph should only ever have one output.

Reply:
it does not hurt to check every output of the graph. I rather check every output?